packages/core/src/odata/common/filter/filter-function.ts

```diff
@@ -34,29 +34,56 @@ export function filterFunction<EntityT extends EntityBase>(
  */
 export function filterFunction<EntityT extends EntityBase>(
   functionName: string,
-  returnType: 'boolean' | 'bool' | 'int' | 'double' | 'decimal' | 'string',
+  returnType: FilterFunctionReturnType,
   ...parameters: FilterFunctionParameterType<EntityT>[]
 ):
   | BooleanFilterFunction<EntityT>
   | NumberFilterFunction<EntityT>
   | StringFilterFunction<EntityT> {
-  if (returnType === 'boolean' || returnType === 'bool') {
-    return new BooleanFilterFunction(functionName, parameters);
-  }
-  if (returnType === 'string') {
-    return new StringFilterFunction(functionName, parameters);
+  return createFilterFunction(functionName, returnType, ...parameters);
+}
+
+// eslint-disable-next-line valid-jsdoc
+/**
+ * @hidden
+ */
+export function createFilterFunction<EntityT extends EntityBase>(
+  functionName: string,
+  returnType: FilterFunctionReturnType,
+  ...parameters: FilterFunctionParameterType<EntityT>[]
+):
+  | BooleanFilterFunction<EntityT>
+  | NumberFilterFunction<EntityT>
+  | StringFilterFunction<EntityT> {
+  switch (returnType) {
```

Comment:
ah finally switches :-)

Reply:
☕️

Reply:
And I agree it is slightly more appropriate here